paddle/optimizer/parameter_optimizer.cc

```diff
@@ -0,0 +1,67 @@
+#include "parameter_optimizer.h"
+#include "optimizer_factory.h"
+#include <glog/logging.h>
+
+namespace paddle {
+namespace optimizer {
+
+template <class T>
+ParameterOptimizer<T>* ParameterOptimizer<T>::create(
+    const ::std::string &config_proto) {
+  paddle::OptimizerConfig config;
+  CHECK(config.ParseFromString(config_proto) == 0) << "error : optimizer config";
+  CHECK(config_valid(config) == 0) << "error : invalid optimizer config ";
+  ParameterOptimizer<T> *opt = nullptr;
+  switch (config.optimizer_name()) {
+  case "SGD" : opt = new SGDOptimizer<T>(config); break;
```

Comment:
I think this factory method should just parse all the config proto, and call constructor of required optimizer. The optimizer should not know anything about protobuffer.